.github/workflows/sync-examples.yml

```diff
@@ -0,0 +1,110 @@
+name: Sync examples
+
+on:
+  workflow_dispatch:
+    inputs:
+      skip-unchanged-check:
+        type: boolean
+        default: false
+      dry-run:
+        type: boolean
+        default: false
+  push:
+    branches:
+      - main
+      - next
+
+# Automatically cancel in-progress actions on the same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  # Allow auto-branch-sync-action to git push
+  contents: write
+
+jobs:
+  check_for_update:
+    # Only allow running on `main` and `next` branch to prevent accidental sync overridding
+    if: ${{ github.repository_owner == 'withastro' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/next') }}
+    name: Check for Updates
+    runs-on: ubuntu-latest
+    outputs:
+      run_job: ${{ steps.check_files.outputs.run_job }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
```

Comment:
This can probably run on a more recent version, not that it makes much difference but it might be slightly faster

Reply:
I've updated the checking flow with a simplified directory check instead which should be a lot faster and doesn't use node